snapcraft_legacy/storeapi/v2/validation_sets.py

```diff
@@ -14,161 +14,119 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from typing import Any, Dict, List, Optional
+from typing import Any, Annotated, Dict, Literal, TYPE_CHECKING
+import pydantic
 
-import jsonschema
+from craft_application import models
 
-from ._api_schema import BUILD_ASSERTION_JSONSCHEMA
 
+if TYPE_CHECKING:
+    SnapName = str
+    SnapId = str
+else:
+    SnapName = pydantic.constr(max_length=40)
+    SnapId = pydantic.constr(max_length=40)
 
-class Snap:
+
+def cast_dict_values_to_strings(data: dict[str, Any]) -> dict[str, Any]:
+    """Cast all scalars in a dictionary to strings."""
+    return {k: _to_string(v) for k, v in data.items()}
+
+
+def _to_string(data: list | str | int | None) -> dict | list | str | None:
+    """Recurse through nested lists and cast all scalars to strings."""
+    if isinstance(data, dict):
+        return {k: _to_string(v) for k, v in data.items()}
```

Comment:
What do you expect the output of `_to_string({1: "1"})` to be?

Reply:
Nice catch.

Even though dictionary keys for these pydantic models will all be strings, I updated this function to cast keys to strings and added some extra tests.